ctapipe/io/dl1eventsource.py

```diff
@@ -0,0 +1,364 @@
+                    dl1.image_mask = image_row["image_mask"]
+                    # ToDo: Find a place in the data container, where the true images can go #1368
+
+                if DataLevel.DL1_PARAMETERS in self.datalevels:
+                    if f"tel_{tel:03d}" not in param_readers.keys():
+                        logger.debug(
+                            f"Triggered telescope {tel} is missing "
+                            "from the parameters table."
+                        )
+                        continue
+                    # Is there a smarter way to unpack this?
+                    # Best would probbaly be if we could directly read
+                    # into the ImageParametersContainer
+                    params = next(param_readers[f"tel_{tel:03d}"])
+                    dl1.parameters.hillas = params[0]
+                    dl1.parameters.timing = params[1]
+                    dl1.parameters.leakage = params[2]
+                    dl1.parameters.concentration = params[3]
+                    dl1.parameters.morphology = params[4]
+                    dl1.parameters.intensity_statistics = params[5]
+                    dl1.parameters.peak_time_statistics = params[6]
+                    # ToDo: Find a place in the data container, where the true params can go #1368
+            yield data
+
+    def _fill_array_pointing(self, data):
+        """
+        Fill the array pointing information of a given event
+        """
+        # Only unique pointings are stored, so reader.read() wont work as easily
+        # Not sure if this is the right way to do it
+        # One could keep an index of the last selected row and the last pointing
+        # and only advance with the row iterator if the next time
+        # is closer or smth like that. But that would require peeking ahead
+        closest_time = np.argmin(
+            self.file_.root.dl1.monitoring.subarray.pointing.col("time")
+            - data.trigger.time
+        )
+        array_pointing = self.file_.root.dl1.monitoring.subarray.pointing[closest_time]
+
+        data.pointing.array_azimuth = u.Quantity(array_pointing["array_azimuth"], u.rad)
+        data.pointing.array_altitude = u.Quantity(
+            array_pointing["array_altitude"], u.rad
+        )
+        data.pointing.array_ra = u.Quantity(array_pointing["array_ra"], u.rad)
+        data.pointing.array_dec = u.Quantity(array_pointing["array_dec"], u.rad)
+
+    def _fill_telescope_pointing(self, data):
+        """
+        Fill the telescope pointing information of a given event
+        """
+        # Same comments as to _fill_array_pointing apply
+        for tel in data.trigger.tel.keys():
+            if self.allowed_tels and tel not in self.allowed_tels:
+                continue
+            tel_pointing_table = self.file_.root.dl1.monitoring.telescope.pointing[
+                f"tel_{tel:03d}"
+            ]
+            closest_time = np.argmin(
```

Comment:
I don't think we should use this approach here.

It will be very expensive to do this for each telescope event.

Maybe better check if there are different pointings and only search for the closest one if there are multiple pointings?

Also, there are data structures that are much better suited to finding something "closest" than calling argmin on the difference.

See for example what I started implementing in #1325

Reply:
> Maybe better check if there are different pointings and only search for the closest one if there are multiple pointings?

:+1: 
 
> Also, there are data structures that are much better suited to finding something "closest" than calling argmin on the difference.
> 
> See for example what I started implementing in #1325

I will need to have a look into that.